testutils/common.py

```diff
@@ -241,23 +247,33 @@ def make_pending_device(dauthd1, dauthm, utoken, tenant_token=""):
     return dev
 
 
-def make_accepted_device(dauthd1, dauthm, utoken, tenant_token=""):
+def make_accepted_device(
+    dauthd1: ApiClient,
+    dauthm: ApiClient,
+    utoken: str,
+    tenant_token: str = "",
+    test_type: str = "regular",
+) -> Device:
+    """Create one device with "accepted" status."""
+    test_types = ["regular", "azure"]
+    if test_type not in test_types:
+        raise RuntimeError("Given test type is not allowed")
     dev = make_pending_device(dauthd1, dauthm, utoken, tenant_token=tenant_token)
     aset_id = dev.authsets[0].id
     change_authset_status(dauthm, dev.id, aset_id, "accepted", utoken)
-
     aset = dev.authsets[0]
     aset.status = "accepted"
 
-    # obtain auth token
-    body, sighdr = deviceauth.auth_req(
-        aset.id_data, aset.pubkey, aset.privkey, tenant_token
-    )
-
-    r = dauthd1.call("POST", deviceauth.URL_AUTH_REQS, body, headers=sighdr)
-
-    assert r.status_code == 200
-    dev.token = r.text
+    # TODO: very bad workaround for Azure IoT Hub backend test; following part is responsible for creating
+    # TODO: additonal, unnecessary auth set which causes Azure test to fail
+    if test_type == "regular":
+        # obtain auth token
+        body, sighdr = deviceauth.auth_req(
+            aset.id_data, aset.pubkey, aset.privkey, tenant_token
+        )
+        r = dauthd1.call("POST", deviceauth.URL_AUTH_REQS, body, headers=sighdr)
+        assert r.status_code == 200
+        dev.token = r.text
```

Comment:
ouch! This is indeed not ideal...

Reply:
I need to fix [this](https://tracker.mender.io/browse/MEN-5281) to be able to remove this awful if :D